// @esri/calcite-components/dist/cjs/calcite-flow-item.cjs.entry.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.10.1
 */
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-26ed993f.js');
const dom = require('./dom-c234ded9.js');
const interactive = require('./interactive-58e7bc5f.js');
const loadable = require('./loadable-2d3795cf.js');
const locale = require('./locale-f5df94e4.js');
const t9n = require('./t9n-7a49ae94.js');
const resources = require('./resources-de9bdb29.js');
require('./guid-0ccaf289.js');
require('./resources-8d1f7e23.js');
require('./browser-0f2eda43.js');
require('./key-404f4cc6.js');
require('./observers-719bb9c5.js');

const CSS = {
    backButton: "back-button",
};
const ICONS = {
    backLeft: "chevron-left",
    backRight: "chevron-right",
};
const SLOTS = {
    actionBar: "action-bar",
    contentTop: "content-top",
    contentBottom: "content-bottom",
    headerActionsStart: "header-actions-start",
    headerActionsEnd: "header-actions-end",
    headerMenuActions: "header-menu-actions",
    headerContent: "header-content",
    fab: "fab",
    footer: "footer",
    footerActions: "footer-actions",
    footerEnd: "footer-end",
    footerStart: "footer-start",
};

const flowItemCss = ":host{box-sizing:border-box;background-color:var(--calcite-color-foreground-1);color:var(--calcite-color-text-2);font-size:var(--calcite-font-size--1)}:host *{box-sizing:border-box}:host([disabled]){cursor:default;-webkit-user-select:none;user-select:none;opacity:var(--calcite-opacity-disabled)}:host([disabled]) *,:host([disabled]) ::slotted(*){pointer-events:none}:host{position:relative;display:flex;inline-size:100%;flex:1 1 auto;overflow:hidden}:host([disabled]) ::slotted([calcite-hydrated][disabled]),:host([disabled]) [calcite-hydrated][disabled]{opacity:1}.interaction-container{display:contents}.back-button{border-width:0px;border-style:solid;border-color:var(--calcite-color-border-3);border-inline-end-width:1px}calcite-panel{--calcite-panel-footer-padding:var(--calcite-flow-item-footer-padding);--calcite-panel-header-border-block-end:var(--calcite-flow-item-header-border-block-end)}:host([hidden]){display:none}[hidden]{display:none}";
const CalciteFlowItemStyle0 = flowItemCss;

const FlowItem = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.calciteFlowItemBack = index.createEvent(this, "calciteFlowItemBack", 7);
        this.calciteFlowItemScroll = index.createEvent(this, "calciteFlowItemScroll", 6);
        this.calciteFlowItemClose = index.createEvent(this, "calciteFlowItemClose", 6);
        this.calciteFlowItemToggle = index.createEvent(this, "calciteFlowItemToggle", 6);
        // --------------------------------------------------------------------------
        //
        //  Private Methods
        //
        // --------------------------------------------------------------------------
        this.handlePanelScroll = (event) => {
            event.stopPropagation();
            this.calciteFlowItemScroll.emit();
        };
        this.handlePanelClose = (event) => {
            event.stopPropagation();
            this.calciteFlowItemClose.emit();
        };
        this.handlePanelToggle = (event) => {
            event.stopPropagation();
            this.collapsed = event.target.collapsed;
            this.calciteFlowItemToggle.emit();
        };
        this.backButtonClick = () => {
            this.calciteFlowItemBack.emit();
        };
        this.setBackRef = (node) => {
            this.backButtonEl = node;
        };
        this.setContainerRef = (node) => {
            this.containerEl = node;
        };
        this.closable = false;
        this.closed = false;
        this.collapsed = false;
        this.collapseDirection = "down";
        this.collapsible = false;
        this.beforeBack = undefined;
        this.description = undefined;
        this.disabled = false;
        this.heading = undefined;
        this.headingLevel = undefined;
        this.loading = false;
        this.menuOpen = false;
        this.messageOverrides = undefined;
        this.messages = undefined;
        this.overlayPositioning = "absolute";
        this.showBackButton = false;
        this.defaultMessages = undefined;
        this.effectiveLocale = "";
    }
    onMessagesChange() {
        /* wired up by t9n util */
    }
    //--------------------------------------------------------------------------
    //
    //  Lifecycle
    //
    //--------------------------------------------------------------------------
    connectedCallback() {
        interactive.connectInteractive(this);
        locale.connectLocalized(this);
        t9n.connectMessages(this);
    }
    async componentWillLoad() {
        await t9n.setUpMessages(this);
        loadable.setUpLoadableComponent(this);
    }
    componentDidRender() {
        interactive.updateHostInteraction(this);
    }
    disconnectedCallback() {
        interactive.disconnectInteractive(this);
        locale.disconnectLocalized(this);
        t9n.disconnectMessages(this);
    }
    componentDidLoad() {
        loadable.setComponentLoaded(this);
    }
    effectiveLocaleChange() {
        t9n.updateMessages(this, this.effectiveLocale);
    }
    // --------------------------------------------------------------------------
    //
    //  Methods
    //
    // --------------------------------------------------------------------------
    /**
     * Sets focus on the component.
     *
     * @returns promise.
     */
    async setFocus() {
        await loadable.componentFocusable(this);
        const { backButtonEl, containerEl } = this;
        if (backButtonEl) {
            return backButtonEl.setFocus();
        }
        else if (containerEl) {
            return containerEl.setFocus();
        }
    }
    /**
     * Scrolls the component's content to a specified set of coordinates.
     *
     * @example
     * myCalciteFlowItem.scrollContentTo({
     *   left: 0, // Specifies the number of pixels along the X axis to scroll the window or element.
     *   top: 0, // Specifies the number of pixels along the Y axis to scroll the window or element
     *   behavior: "auto" // Specifies whether the scrolling should animate smoothly (smooth), or happen instantly in a single jump (auto, the default value).
     * });
     * @param options - allows specific coordinates to be defined.
     * @returns - promise that resolves once the content is scrolled to.
     */
    async scrollContentTo(options) {
        await this.containerEl?.scrollContentTo(options);
    }
    // --------------------------------------------------------------------------
    //
    //  Render Methods
    //
    // --------------------------------------------------------------------------
    renderBackButton() {
        const { el } = this;
        const rtl = dom.getElementDir(el) === "rtl";
        const { showBackButton, backButtonClick, messages } = this;
        const label = messages.back;
        const icon = rtl ? ICONS.backRight : ICONS.backLeft;
        return showBackButton ? (index.h("calcite-action", { "aria-label": label, class: CSS.backButton, icon: icon, key: "flow-back-button", onClick: backButtonClick, ref: this.setBackRef, scale: "s", slot: "header-actions-start", text: label, title: label })) : null;
    }
    render() {
        const { collapsed, collapseDirection, collapsible, closable, closed, description, disabled, heading, headingLevel, loading, menuOpen, messages, overlayPositioning, } = this;
        return (index.h(index.Host, { key: '386f2c2d780c8b3a1be28a8680b6f99c9ded37e6' }, index.h(interactive.InteractiveContainer, { key: '6e80e71a06923e966ac75f8b8c172586444e3ccb', disabled: disabled }, index.h("calcite-panel", { key: '175b4bcfdecb5b4c59d5c76db6fcae7975ee5884', closable: closable, closed: closed, collapseDirection: collapseDirection, collapsed: collapsed, collapsible: collapsible, description: description, disabled: disabled, heading: heading, headingLevel: headingLevel, loading: loading, menuOpen: menuOpen, messageOverrides: messages, onCalcitePanelClose: this.handlePanelClose, onCalcitePanelScroll: this.handlePanelScroll, onCalcitePanelToggle: this.handlePanelToggle, overlayPositioning: overlayPositioning, ref: this.setContainerRef }, this.renderBackButton(), index.h("slot", { key: '6e0ac6b3c3b62a5be7a6a21fbe55e56c1d4ad90a', name: SLOTS.actionBar, slot: resources.SLOTS.actionBar }), index.h("slot", { key: 'b2036686e62d9b5c8ecc39e8ee025e98798d7dec', name: SLOTS.headerActionsStart, slot: resources.SLOTS.headerActionsStart }), index.h("slot", { key: '66c09bbb9cd21327a9cc04b3046c5264b475a77d', name: SLOTS.headerActionsEnd, slot: resources.SLOTS.headerActionsEnd }), index.h("slot", { key: 'cd02bc580db5ecdd98b0bbbfa99655871c04bda4', name: SLOTS.headerContent, slot: resources.SLOTS.headerContent }), index.h("slot", { key: 'b855684ce946c66b4c13756abd1865ab420715f0', name: SLOTS.headerMenuActions, slot: resources.SLOTS.headerMenuActions }), index.h("slot", { key: 'f1d29575abbdb582c04d674dfef7f4a9d385da10', name: SLOTS.fab, slot: resources.SLOTS.fab }), index.h("slot", { key: '3033128596131e0f1fd62b28cba07957d1707d0c', name: SLOTS.contentTop, slot: resources.SLOTS.contentTop }), index.h("slot", { key: 'c8cf03fbacebe288afc909b9c8989242c201df8c', name: SLOTS.contentBottom, slot: resources.SLOTS.contentBottom }), index.h("slot", { key: '09e19f66b3a855689539dc511b1b826830a90283', name: SLOTS.footer, slot: resources.SLOTS.footer }, index.h("slot", { key: 'e7519384a44ee5f1f19d9747c44007ca7b95f4c8', name: SLOTS.footerStart, slot: resources.SLOTS.footerStart }), index.h("slot", { key: 'abd54ab99d62783f603118361bda307c3d108af0', name: SLOTS.footerEnd, slot: resources.SLOTS.footerEnd })), index.h("slot", { key: '9319eff4559e39701a1f838c1add5b8f57f01f2b', name: SLOTS.footerActions, slot: resources.SLOTS.footerActions }), index.h("slot", { key: 'bd1cc6da3068868bfaf3ba8a0e524d0d138db940' })))));
    }
    static get assetsDirs() { return ["assets"]; }
    get el() { return index.getElement(this); }
    static get watchers() { return {
        "messageOverrides": ["onMessagesChange"],
        "effectiveLocale": ["effectiveLocaleChange"]
    }; }
};
FlowItem.style = CalciteFlowItemStyle0;

exports.calcite_flow_item = FlowItem;
