// @esri/calcite-components/dist/cjs/fr-ch-a0a67a45.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.10.1
 */
'use strict';

const calciteInputTimePicker_calciteTimePicker_entry = require('./calcite-input-time-picker.calcite-time-picker-77df3dd4.js');
require('./index-26ed993f.js');
require('./form-e70bd946.js');
require('./dom-c234ded9.js');
require('./guid-0ccaf289.js');
require('./resources-8d1f7e23.js');
require('./interactive-58e7bc5f.js');
require('./browser-0f2eda43.js');
require('./key-404f4cc6.js');
require('./label-f834fb76.js');
require('./component-9dc22f29.js');
require('./loadable-2d3795cf.js');
require('./locale-f5df94e4.js');
require('./observers-719bb9c5.js');
require('./focusTrapComponent-fb846fc9.js');
require('./config-4960d556.js');
require('./math-a7d4d32f.js');
require('./t9n-7a49ae94.js');
require('./openCloseComponent-21bb3aad.js');
require('./Validation-a01c24c2.js');
require('./input-cf688991.js');

// French (Switzerland) [fr-ch]
var locale = {
  name: 'fr-ch',
  weekdays: 'dimanche_lundi_mardi_mercredi_jeudi_vendredi_samedi'.split('_'),
  months: 'janvier_février_mars_avril_mai_juin_juillet_août_septembre_octobre_novembre_décembre'.split('_'),
  weekStart: 1,
  weekdaysShort: 'dim._lun._mar._mer._jeu._ven._sam.'.split('_'),
  monthsShort: 'janv._févr._mars_avr._mai_juin_juil._août_sept._oct._nov._déc.'.split('_'),
  weekdaysMin: 'di_lu_ma_me_je_ve_sa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY HH:mm',
    LLLL: 'dddd D MMMM YYYY HH:mm'
  },
  relativeTime: {
    future: 'dans %s',
    past: 'il y a %s',
    s: 'quelques secondes',
    m: 'une minute',
    mm: '%d minutes',
    h: 'une heure',
    hh: '%d heures',
    d: 'un jour',
    dd: '%d jours',
    M: 'un mois',
    MM: '%d mois',
    y: 'un an',
    yy: '%d ans'
  }
};
calciteInputTimePicker_calciteTimePicker_entry.dayjs.locale(locale, null, true);

exports['default'] = locale;
