// @esri/calcite-components/dist/cjs/id-d9aeb01c.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.10.1
 */
'use strict';

const calciteInputTimePicker_calciteTimePicker_entry = require('./calcite-input-time-picker.calcite-time-picker-77df3dd4.js');
require('./index-26ed993f.js');
require('./form-e70bd946.js');
require('./dom-c234ded9.js');
require('./guid-0ccaf289.js');
require('./resources-8d1f7e23.js');
require('./interactive-58e7bc5f.js');
require('./browser-0f2eda43.js');
require('./key-404f4cc6.js');
require('./label-f834fb76.js');
require('./component-9dc22f29.js');
require('./loadable-2d3795cf.js');
require('./locale-f5df94e4.js');
require('./observers-719bb9c5.js');
require('./focusTrapComponent-fb846fc9.js');
require('./config-4960d556.js');
require('./math-a7d4d32f.js');
require('./t9n-7a49ae94.js');
require('./openCloseComponent-21bb3aad.js');
require('./Validation-a01c24c2.js');
require('./input-cf688991.js');

// Indonesian [id]
var locale = {
  name: 'id',
  weekdays: 'Minggu_Senin_Selasa_Rabu_Kamis_Jumat_Sabtu'.split('_'),
  months: 'Januari_Februari_Maret_April_Mei_Juni_Juli_Agustus_September_Oktober_November_Desember'.split('_'),
  weekdaysShort: 'Min_Sen_Sel_Rab_Kam_Jum_Sab'.split('_'),
  monthsShort: 'Jan_Feb_Mar_Apr_Mei_Jun_Jul_Agt_Sep_Okt_Nov_Des'.split('_'),
  weekdaysMin: 'Mg_Sn_Sl_Rb_Km_Jm_Sb'.split('_'),
  weekStart: 1,
  formats: {
    LT: 'HH.mm',
    LTS: 'HH.mm.ss',
    L: 'DD/MM/YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY [pukul] HH.mm',
    LLLL: 'dddd, D MMMM YYYY [pukul] HH.mm'
  },
  relativeTime: {
    future: 'dalam %s',
    past: '%s yang lalu',
    s: 'beberapa detik',
    m: 'semenit',
    mm: '%d menit',
    h: 'sejam',
    hh: '%d jam',
    d: 'sehari',
    dd: '%d hari',
    M: 'sebulan',
    MM: '%d bulan',
    y: 'setahun',
    yy: '%d tahun'
  },
  ordinal: function ordinal(n) {
    return n + ".";
  }
};
calciteInputTimePicker_calciteTimePicker_entry.dayjs.locale(locale, null, true);

exports['default'] = locale;
