// @esri/calcite-components/dist/cjs/mk-5ca4ea7e.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.10.1
 */
'use strict';

const calciteInputTimePicker_calciteTimePicker_entry = require('./calcite-input-time-picker.calcite-time-picker-77df3dd4.js');
require('./index-26ed993f.js');
require('./form-e70bd946.js');
require('./dom-c234ded9.js');
require('./guid-0ccaf289.js');
require('./resources-8d1f7e23.js');
require('./interactive-58e7bc5f.js');
require('./browser-0f2eda43.js');
require('./key-404f4cc6.js');
require('./label-f834fb76.js');
require('./component-9dc22f29.js');
require('./loadable-2d3795cf.js');
require('./locale-f5df94e4.js');
require('./observers-719bb9c5.js');
require('./focusTrapComponent-fb846fc9.js');
require('./config-4960d556.js');
require('./math-a7d4d32f.js');
require('./t9n-7a49ae94.js');
require('./openCloseComponent-21bb3aad.js');
require('./Validation-a01c24c2.js');
require('./input-cf688991.js');

// Macedonian [mk]
var locale = {
  name: 'mk',
  weekdays: 'недела_понеделник_вторник_среда_четврток_петок_сабота'.split('_'),
  months: 'јануари_февруари_март_април_мај_јуни_јули_август_септември_октомври_ноември_декември'.split('_'),
  weekStart: 1,
  weekdaysShort: 'нед_пон_вто_сре_чет_пет_саб'.split('_'),
  monthsShort: 'јан_фев_мар_апр_мај_јун_јул_авг_сеп_окт_ное_дек'.split('_'),
  weekdaysMin: 'нe_пo_вт_ср_че_пе_сa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'D.MM.YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY H:mm',
    LLLL: 'dddd, D MMMM YYYY H:mm'
  },
  relativeTime: {
    future: 'после %s',
    past: 'пред %s',
    s: 'неколку секунди',
    m: 'минута',
    mm: '%d минути',
    h: 'час',
    hh: '%d часа',
    d: 'ден',
    dd: '%d дена',
    M: 'месец',
    MM: '%d месеци',
    y: 'година',
    yy: '%d години'
  }
};
calciteInputTimePicker_calciteTimePicker_entry.dayjs.locale(locale, null, true);

exports['default'] = locale;
