// @esri/calcite-components/dist/collection/components/option-group/option-group.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.10.1
 */
import { Fragment, h } from "@stencil/core";
/**
 * @slot - A slot for adding `calcite-option`s.
 */
export class OptionGroup {
    constructor() {
        this.disabled = false;
        this.label = undefined;
    }
    handlePropChange() {
        this.calciteInternalOptionGroupChange.emit();
    }
    //--------------------------------------------------------------------------
    //
    //  Render Methods
    //
    //--------------------------------------------------------------------------
    render() {
        return (h(Fragment, { key: 'b4cd175214829ea1d1751ddae82e40559e477867' }, h("div", { key: '5342ad95ad2f3e7ba075bc5fb12e51b937f221af' }, this.label), h("slot", { key: '8cd0822a67680aec1dcebaba2b64a7c8389a1503' })));
    }
    static get is() { return "calcite-option-group"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["option-group.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["option-group.css"]
        };
    }
    static get properties() {
        return {
            "disabled": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "When `true`, interaction is prevented and the component is displayed with lower opacity."
                },
                "attribute": "disabled",
                "reflect": true,
                "defaultValue": "false"
            },
            "label": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": true,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": "Accessible name for the component."
                },
                "attribute": "label",
                "reflect": false
            }
        };
    }
    static get events() {
        return [{
                "method": "calciteInternalOptionGroupChange",
                "name": "calciteInternalOptionGroupChange",
                "bubbles": true,
                "cancelable": false,
                "composed": true,
                "docs": {
                    "tags": [{
                            "name": "internal",
                            "text": undefined
                        }],
                    "text": ""
                },
                "complexType": {
                    "original": "void",
                    "resolved": "void",
                    "references": {}
                }
            }];
    }
    static get watchers() {
        return [{
                "propName": "disabled",
                "methodName": "handlePropChange"
            }, {
                "propName": "label",
                "methodName": "handlePropChange"
            }];
    }
}
