// @deck.gl/carto/dist/index.js
import { default as ClusterTileLayer } from "./layers/cluster-tile-layer.js";
import { default as H3TileLayer } from "./layers/h3-tile-layer.js";
import { default as HeatmapTileLayer } from "./layers/heatmap-tile-layer.js";
import { default as _PointLabelLayer } from "./layers/point-label-layer.js";
import { default as QuadbinTileLayer } from "./layers/quadbin-tile-layer.js";
import { default as RasterTileLayer } from "./layers/raster-tile-layer.js";
import { default as VectorTileLayer } from "./layers/vector-tile-layer.js";
const CARTO_LAYERS = {
    ClusterTileLayer,
    H3TileLayer,
    HeatmapTileLayer,
    _PointLabelLayer,
    QuadbinTileLayer,
    RasterTileLayer,
    VectorTileLayer
};
export { CARTO_LAYERS, ClusterTileLayer, H3TileLayer, HeatmapTileLayer, _PointLabelLayer, QuadbinTileLayer, RasterTileLayer, VectorTileLayer };
export { default as BASEMAP, GOOGLE_BASEMAPS as _GOOGLE_BASEMAPS, getStyleUrl as _getStyleUrl, fetchStyle as _fetchStyle, applyLayerGroupFilters as _applyLayerGroupFilters, STYLE_LAYER_GROUPS as _STYLE_LAYER_GROUPS } from "./basemap.js";
export { default as colorBins } from "./style/color-bins-style.js";
export { default as colorCategories } from "./style/color-categories-style.js";
export { default as colorContinuous } from "./style/color-continuous-style.js";
export { CartoAPIError, fetchMap, query } from "./api/index.js";
export { fetchBasemapProps } from "./api/basemap.js";
import { boundaryQuerySource, boundaryTableSource, h3QuerySource, h3TableSource, h3TilesetSource, rasterSource, quadbinQuerySource, quadbinTableSource, quadbinTilesetSource, vectorQuerySource, vectorTableSource, vectorTilesetSource, SOURCE_DEFAULTS } from "./sources/index.js";
const CARTO_SOURCES = {
    boundaryQuerySource,
    boundaryTableSource,
    h3QuerySource,
    h3TableSource,
    h3TilesetSource,
    rasterSource,
    quadbinQuerySource,
    quadbinTableSource,
    quadbinTilesetSource,
    vectorQuerySource,
    vectorTableSource,
    vectorTilesetSource
};
export { boundaryQuerySource, boundaryTableSource, h3QuerySource, h3TableSource, h3TilesetSource, rasterSource, quadbinQuerySource, quadbinTableSource, quadbinTilesetSource, vectorQuerySource, vectorTableSource, vectorTilesetSource, CARTO_SOURCES, SOURCE_DEFAULTS };
