// @esri/calcite-components/dist/esm/el-213c1f06.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.10.1
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-6f6c19e0.js';
import './index-c69832bf.js';
import './form-326c9435.js';
import './dom-5badf0bf.js';
import './guid-13f267a8.js';
import './resources-fbc9037f.js';
import './interactive-81282a6f.js';
import './browser-c7ad9bc1.js';
import './key-3685937e.js';
import './label-e3157659.js';
import './component-f1a729cc.js';
import './loadable-1c82fe7c.js';
import './locale-2274ef15.js';
import './observers-9479c046.js';
import './focusTrapComponent-a4d396cc.js';
import './config-cdc3bb2b.js';
import './math-1c4eaaa4.js';
import './t9n-2ba79023.js';
import './openCloseComponent-2e1b5522.js';
import './Validation-04eac16d.js';
import './input-fe03bb7b.js';

// Greek [el]
var locale = {
  name: 'el',
  weekdays: 'Κυριακή_Δευτέρα_Τρίτη_Τετάρτη_Πέμπτη_Παρασκευή_Σάββατο'.split('_'),
  weekdaysShort: 'Κυρ_Δευ_Τρι_Τετ_Πεμ_Παρ_Σαβ'.split('_'),
  weekdaysMin: 'Κυ_Δε_Τρ_Τε_Πε_Πα_Σα'.split('_'),
  months: 'Ιανουάριος_Φεβρουάριος_Μάρτιος_Απρίλιος_Μάιος_Ιούνιος_Ιούλιος_Αύγουστος_Σεπτέμβριος_Οκτώβριος_Νοέμβριος_Δεκέμβριος'.split('_'),
  monthsShort: 'Ιαν_Φεβ_Μαρ_Απρ_Μαι_Ιουν_Ιουλ_Αυγ_Σεπτ_Οκτ_Νοε_Δεκ'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  weekStart: 1,
  relativeTime: {
    future: 'σε %s',
    past: 'πριν %s',
    s: 'μερικά δευτερόλεπτα',
    m: 'ένα λεπτό',
    mm: '%d λεπτά',
    h: 'μία ώρα',
    hh: '%d ώρες',
    d: 'μία μέρα',
    dd: '%d μέρες',
    M: 'ένα μήνα',
    MM: '%d μήνες',
    y: 'ένα χρόνο',
    yy: '%d χρόνια'
  },
  formats: {
    LT: 'h:mm A',
    LTS: 'h:mm:ss A',
    L: 'DD/MM/YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY h:mm A',
    LLLL: 'dddd, D MMMM YYYY h:mm A'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
