// @esri/calcite-components/dist/esm/it-ch-33b5479f.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.10.1
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-6f6c19e0.js';
import './index-c69832bf.js';
import './form-326c9435.js';
import './dom-5badf0bf.js';
import './guid-13f267a8.js';
import './resources-fbc9037f.js';
import './interactive-81282a6f.js';
import './browser-c7ad9bc1.js';
import './key-3685937e.js';
import './label-e3157659.js';
import './component-f1a729cc.js';
import './loadable-1c82fe7c.js';
import './locale-2274ef15.js';
import './observers-9479c046.js';
import './focusTrapComponent-a4d396cc.js';
import './config-cdc3bb2b.js';
import './math-1c4eaaa4.js';
import './t9n-2ba79023.js';
import './openCloseComponent-2e1b5522.js';
import './Validation-04eac16d.js';
import './input-fe03bb7b.js';

// Italian (Switzerland) [it-ch]
var locale = {
  name: 'it-ch',
  weekdays: 'domenica_lunedì_martedì_mercoledì_giovedì_venerdì_sabato'.split('_'),
  months: 'gennaio_febbraio_marzo_aprile_maggio_giugno_luglio_agosto_settembre_ottobre_novembre_dicembre'.split('_'),
  weekStart: 1,
  weekdaysShort: 'dom_lun_mar_mer_gio_ven_sab'.split('_'),
  monthsShort: 'gen_feb_mar_apr_mag_giu_lug_ago_set_ott_nov_dic'.split('_'),
  weekdaysMin: 'do_lu_ma_me_gi_ve_sa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY HH:mm',
    LLLL: 'dddd D MMMM YYYY HH:mm'
  },
  relativeTime: {
    future: 'tra %s',
    past: '%s fa',
    s: 'alcuni secondi',
    m: 'un minuto',
    mm: '%d minuti',
    h: 'un\'ora',
    hh: '%d ore',
    d: 'un giorno',
    dd: '%d giorni',
    M: 'un mese',
    MM: '%d mesi',
    y: 'un anno',
    yy: '%d anni'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
