// @esri/calcite-components/dist/esm/ja-60613101.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/dev/LICENSE.md for details.
 * v2.10.1
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-6f6c19e0.js';
import './index-c69832bf.js';
import './form-326c9435.js';
import './dom-5badf0bf.js';
import './guid-13f267a8.js';
import './resources-fbc9037f.js';
import './interactive-81282a6f.js';
import './browser-c7ad9bc1.js';
import './key-3685937e.js';
import './label-e3157659.js';
import './component-f1a729cc.js';
import './loadable-1c82fe7c.js';
import './locale-2274ef15.js';
import './observers-9479c046.js';
import './focusTrapComponent-a4d396cc.js';
import './config-cdc3bb2b.js';
import './math-1c4eaaa4.js';
import './t9n-2ba79023.js';
import './openCloseComponent-2e1b5522.js';
import './Validation-04eac16d.js';
import './input-fe03bb7b.js';

// Japanese [ja]
var locale = {
  name: 'ja',
  weekdays: '日曜日_月曜日_火曜日_水曜日_木曜日_金曜日_土曜日'.split('_'),
  weekdaysShort: '日_月_火_水_木_金_土'.split('_'),
  weekdaysMin: '日_月_火_水_木_金_土'.split('_'),
  months: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),
  monthsShort: '1月_2月_3月_4月_5月_6月_7月_8月_9月_10月_11月_12月'.split('_'),
  ordinal: function ordinal(n) {
    return n + "\u65E5";
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'YYYY/MM/DD',
    LL: 'YYYY年M月D日',
    LLL: 'YYYY年M月D日 HH:mm',
    LLLL: 'YYYY年M月D日 dddd HH:mm',
    l: 'YYYY/MM/DD',
    ll: 'YYYY年M月D日',
    lll: 'YYYY年M月D日 HH:mm',
    llll: 'YYYY年M月D日(ddd) HH:mm'
  },
  meridiem: function meridiem(hour) {
    return hour < 12 ? '午前' : '午後';
  },
  relativeTime: {
    future: '%s後',
    past: '%s前',
    s: '数秒',
    m: '1分',
    mm: '%d分',
    h: '1時間',
    hh: '%d時間',
    d: '1日',
    dd: '%d日',
    M: '1ヶ月',
    MM: '%dヶ月',
    y: '1年',
    yy: '%d年'
  }
};
dayjs.locale(locale, null, true);

export { locale as default };
